### PR TITLE
Link createLonLatToCartesian in `@rapidsai/cuspatial`

### LINKED ISSUE #441: Create lonLatToCartesian to improve visual correspondence of map and NAD coordinates for demo.
I need to add lonLatToCartesian to get better display coordinates. This PR is ready in https://github.com/rapidsai/node/pull/440.

### PR DESCRIPTION
**Pull request by thomcom**
This PR makes the lon_lat_to_cartesian function in cuspatial available in `node-rapids`, including one basic test.

Closes #441

### CODE REVIEW

**Comment by thomcom** [comment]
rerun tests

File: modules/cuspatial/src/addon.cpp [hunk 1]
```diff
@@ -21,19 +21,20 @@ struct rapidsai_cuspatial : public nv::EnvLocalAddon, public Napi::Addon<rapidsa
...
+         "findQuadtreeAndBoundingBoxIntersections"),
+       InstanceMethod<&rapidsai_cuspatial::find_points_in_polygons>("findPointsInPolygons"),
+       InstanceMethod<&rapidsai_cuspatial::find_polyline_nearest_to_each_point>(
+         "findPolylineNearestToEachPoint"),
+       InstanceMethod<&rapidsai_cuspatial::compute_polygon_bounding_boxes>(
+         "computePolygonBoundingBoxes"),
+       InstanceMethod<&rapidsai_cuspatial::compute_polyline_bounding_boxes>(
+         "computePolylineBoundingBoxes"),
+       InstanceMethod<&rapidsai_cuspatial::lonlat_to_cartesian>("lonLatToCartesian")
+
```
Comment (trxcllnt) [inline modules/cuspatial/src/addon.cpp:37]: This seems to have been reformatted with a different version of clang-format than is inside the dev container?

File: modules/cuspatial/src/node_cuspatial.ts [hunk 2]
```diff
@@ -12,12 +12,16 @@
...
+import {Column, FloatingPoint, Int32, Series, Table, Uint32} from '@rapidsai/cudf';
 import {MemoryResource} from '@rapidsai/rmm';
 
 /** @ignore */
 export declare const _cpp_exports: any;
 
+export type SeriesPair<T extends FloatingPoint> = {
+  x: Series<T>; y: Series<T>;
+}
+
```
Comment (trxcllnt) [inline modules/cuspatial/src/node_cuspatial.ts:24]: This type name doesn't seem super descriptive... Can we define it inline? Or rewrite it to use the `Points` type in [geometry.ts](https://github.com/rapidsai/node/blob/2829f0ae92e501eeed192144d049de3a75385ec2/modules/cuspatial/src/geometry.ts#L25)?

File: modules/cuspatial/src/spatial.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import {
+  lonLatToCartesian,
+} from './addon';
+import {
+  SeriesPair,
+} from './node_cuspatial';
+
+export function convertLonLatToCartesian<T extends FloatingPoint>(
+  centerX: number, centerY: number, lonPoints: Column<T>, latPoints: Column<T>): SeriesPair<T> {
```
Comment (trxcllnt) [inline modules/cuspatial/src/spatial.ts:29]:
```suggestion
  centerX: number, centerY: number, lonPoints: Column<T>, latPoints: Column<T>): Series<Point<T>> {
```

File: modules/cuspatial/test/spatial-tests.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2021, NVIDIA CORPORATION.
```
Comment (trxcllnt) [inline modules/cuspatial/test/spatial-tests.ts:1]:
```suggestion
// Copyright (c) 2022, NVIDIA CORPORATION.
```

File: modules/cuspatial/src/spatial.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  const result = lonLatToCartesian(centerX, centerY, lonPoints, latPoints);
+  const x      = Series.new(result.x);
+  const y      = Series.new(result.y);
+  return {x: x, y: y};
```
Comment (trxcllnt) [inline modules/cuspatial/src/spatial.ts:33]:
```suggestion
  return makePoints(x, y);
```

File: modules/cuspatial/src/spatial.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2021-2022, NVIDIA CORPORATION.
```
Comment (trxcllnt) [inline modules/cuspatial/src/spatial.ts:1]:
```suggestion
// Copyright (c) 2022, NVIDIA CORPORATION.
```

File: modules/cuspatial/src/addon.cpp (hunk 1 above)
Comment (thomcom) [reply modules/cuspatial/src/addon.cpp:37]: I noticed this. In the container I have `clang-format version 12.0.1`, which is formatting it this way.

File: modules/cuspatial/src/spatial.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import {makePoints} from '@rapidsai/cuspatial';
+import {Points} from '@rapidsai/cuspatial/geometry';
+
+import {
+  lonLatToCartesian,
+} from './addon';
+
+export function convertLonLatToCartesian<T extends FloatingPoint>(
+  centerX: number, centerY: number, lonPoints: Column<T>, latPoints: Column<T>): Points<T> {
+  const result = lonLatToCartesian(centerX, centerY, lonPoints, latPoints);
```
Comment (trxcllnt) [inline modules/cuspatial/src/spatial.ts:29]:
```suggestion
export function convertLonLatToCartesian<T extends FloatingPoint>(
  centerX: number, centerY: number, lonPoints: Series<T>, latPoints: Series<T>): Points<T> {
  const result = lonLatToCartesian(centerX, centerY, lonPoints._col, latPoints._col);
```

File: modules/cuspatial/test/spatial-tests.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+import {convertLonLatToCartesian} from '@rapidsai/cuspatial';
+import {DeviceBuffer} from '@rapidsai/rmm';
+
+setDefaultAllocator((byteLength: number) => new DeviceBuffer(byteLength));
+
+describe('Spatial', () => {
+  test(`convertLonLatToCartesian`, () => {
+    const seriesX = Series.new([-120.0, -120.0]);
+    const seriesY = Series.new([48.0, 49.0]);
+    const result  = convertLonLatToCartesian(1.0, 1.0, seriesX._col, seriesY._col);
```
Comment (trxcllnt) [inline modules/cuspatial/test/spatial-tests.ts:28]:
```suggestion
    const result  = convertLonLatToCartesian(1.0, 1.0, seriesX, seriesY);
```

File: modules/cuspatial/src/node_cuspatial.ts [hunk 9]
```diff
@@ -78,3 +78,9 @@ export declare function findPolylineNearestToEachPoint<T extends FloatingPoint>(
   polylinePointsY: Column<T>,
   memoryResource
   ?: MemoryResource): {table: Table, names: ['point_index', 'polyline_index', 'distance']};
+
+export declare function lonLatToCartesian(origin_lon: number,
+                                          origin_lat: number,
+                                          lats: Series,
+                                          lons: Series,
+                                          memoryResource?: MemoryResource): {x: Series, y: Series};
```
Comment (trxcllnt) [inline modules/cuspatial/src/node_cuspatial.ts:86]:
```suggestion
export declare function lonLatToCartesian<T extends FloatingPoint>(origin_lon: number,
                                          origin_lat: number,
                                          lats: Column<T>,
                                          lons: Column<T>,
                                          memoryResource?: MemoryResource): {x: Column<T>, y: Column<T>};
```

**Review by trxcllnt — CHANGES_REQUESTED** [review]
Just a few more suggestions, mostly around ensuring typings align with the implementation.

**Comment by thomcom** [comment]
I committed them.

**Review by trxcllnt — APPROVED** [review]

**Pull request merged**